app/src/main/java/me/hufman/androidautoidrive/music/spotify/SpotifyWebApi.kt

```diff
@@ -56,8 +59,8 @@ class SpotifyWebApi private constructor(val context: Context, val appSettings: M
 
 	private val authStateManager: SpotifyAuthStateManager
 	private val clientId: String
-	private var getLikedSongsAttempted: Boolean = false
-	private var spotifyAppControllerCaller: SpotifyAppController? = null
+	private var lastFailedQueueMetadataCreate: (() -> Unit)? = null
```

Comment:
Hmmm how about `pendingQueueMetadataCreate`? I really like the idea of encapsulating the closure with the controller!

Reply:
I like that name much better, updated! Yeah this was the cleanest way I could think of for adding a retry system that way we don't need to store extra information we could just encapsulate the QueueMetadata creation as a method that could be called in the case of Spotify Web API failure